packages/sitecore-jss/src/component-library/utils.ts

```diff
@@ -0,0 +1,93 @@
+import { ComponentRendering, Field, GenericFieldValue } from '../layout/models';
+
+export const COMPONENT_LIBRARY_READY_MESSAGE = { name: 'component:status', message: 'ready' };
+
+export interface ComponentUpdateEventArgs {
+  uid: string;
+  params: Record<string, string>;
+  fields: Record<string, Field<GenericFieldValue>>;
+}
+/**
+ * Adds the browser-side event handler for 'component:update' message used in Component Library
+ * The event should update a component on page by uid, with fields and params from event args
+ * @param {ComponentRendering} rootComponent root component displayed for Component Library page
+ * @param {Function} successCallback  callback to be called after successful component update
+ */
+export const addComponentUpdateHandler = (
+  rootComponent: ComponentRendering,
+  successCallback?: () => void
+) => {
+  if (!window) return;
+  window.addEventListener('message', (e) =>
+    updateComponentHandler(e, rootComponent, successCallback)
+  );
+};
+
+const validateOrigin = (event: MessageEvent) => {
+  // TODO: use `EDITING_ALLOWED_ORIGINS.concat(getAllowedOriginsFromEnv())` later
+  // nextjs's JSS_ALLOWED_ORIGINS is not available on the client, need to use NEXT_PUBLIC_ variable, but it's a breaking change for Deploy
+  const allowedOrigins = ['*'];
+  return allowedOrigins.some(
+    (origin) =>
+      origin === event.origin ||
+      new RegExp('^' + origin.replace('.', '\\.').replace(/\*/g, '.*') + '$').test(event.origin)
+  );
+};
+
+export const updateComponentHandler = (
+  e: MessageEvent,
```

Comment:
considering that postmessage doesn't have a custom name for event lets agree on a structure, so we can clearly separate it, e.g.:
```
{
  name: "component:update",
  data: {
    uid: "",
   params: "",
   fields: "",
} 
```
feel free to suggest  other names if you have preferences

Reply:
The reason "name" args is under "data" is due to the `window.postMessage` signature and `MessageEvent` implementation (https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage). Everything you pass as `message` arg into `postMessage` will be available as `event.data`. 
We can't put `name` outside of `data`, and I can add it to `ComponentUpdateEventArgs` - will that be OK?

Reply:
No you typically want to separate event name from payload. `postMessage` and `MessageEvent` here is just a transport for our custom events. Consider there is a pub-sub abstraction over `postMessage` and have multiple different event's. In such case you need to verify event name first just to identify what it is, before checking a payload. 
And the only problem `MessageEvent` is creating for suggested structure is clunky api when you use it directly:
```
messageEvent.data.name
messageEvent.data.data
```
Which is again problem of a wrapper, and can be easily mitigated by choosing another name for data property, e.g. `detail` like in the `CustomEvent`